ui/packages/consul-ui/app/serializers/nspace.js

```diff
@@ -6,62 +6,69 @@ export default class NspaceSerializer extends Serializer {
   primaryKey = PRIMARY_KEY;
   slugKey = SLUG_KEY;
 
-  respondForQuery(respond, serialized, data) {
-    return respond((headers, body) => {
-      return this.attachHeaders(
-        headers,
-        body.map(function(item) {
-          if (get(item, 'ACLs.PolicyDefaults')) {
-            item.ACLs.PolicyDefaults = item.ACLs.PolicyDefaults.map(function(item) {
-              item.template = '';
+  respondForQuery(respond, query, data, modelClass) {
```

Comment:
I added `item.Namespace = item.Name` and also `item.Datacenter = query.dc`, for keying purposes. Also, now nspaces need to be keyed by DC, I moved it back to use the parent method so that it goes through the apps uid extra fingerprinting like the rest of the app (previously it didn't need the extra keying as namespaces used to be a top level bucket). The rest is just indenting.